kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/LocallyRunningOperatorRbacHandler.java

```diff
@@ -249,5 +414,88 @@ public interface TestActor {
 
         @NonNull
         <T extends HasMetadata> NonNamespaceOperation<T, KubernetesResourceList<T>, Resource<T>> resources(@NonNull Class<T> type);
+
+        default ResourceBuilder<KafkaProxySpecBuilder, KafkaProxySpec, KafkaProxy> kafkaProxy() {
+            return kafkaProxy(randomName());
+        }
+
+        default ResourceBuilder<KafkaProxySpecBuilder, KafkaProxySpec, KafkaProxy> kafkaProxy(String name) {
```

Comment:
I added these methods to access resources by name to make sure we `get` the resource from k8s before attempting to `replace` because some tests were throwing up exceptions about the resource being modified. It's a bit clunky.

Reply:
I've seen those failures too, so I agree it's a problem worth addressing.

This is making the assumption that the resource doesn't change between here and when `replace()` gets called. There's no guarantee that callers will always do a `kafkaProxy(NAME)....replace()` in a single statement though. So it would be a slightly safer (but still not completely safe) assumption if you deferred the `get` till when you do the `replace`.

Reply:
yeahp, the code is not too far from being able to optimistically retry and apply the user's desired edits